internal/generator/templates/decode_compound_parameters.go.tmpl

```diff
@@ -1,28 +1,29 @@
 
 {{- $scope := .pkg }}
 {{- $field := .field }}
+{{- $native := .native }}
 {{- $field_name := .field_name }}
 {{- $use_index_operator := hasIndexOperators $field.Type }}
 {{- if $field.Type.TypeArguments }}
     {{- if $field.Array }}
-        {{- if $use_index_operator }}
-// This array uses the index operator. The following for loop passes
-// all parameters to individual default objects. Since the @index
-// operator is used, each default object entry may look differently.
-for index, defaultObject := range {{ $field.Name }}ArrayProperties.ArrayTraits.ObjectCreator.DefaultObjects {
-          {{- $field_name = "defaultObject" }}
-        {{- else }}
-          {{- $field_name = printf "%sArrayProperties.ArrayTraits.ObjectCreator.DefaultObject" $field.Name }}
-        {{- end }}
+        {{ $field.Name }}ArrayProperties.SetCompoundParameterFn = func(index int, elem *{{ goType $scope $native.Type }}) {
+     {{- end}}
+    {{- if $use_index_operator }}
+    // This array uses the index operator. The following for loop passes
+    // all parameters to individual default objects. Since the @index
+    // operator is used, each default object entry may look differently.
     {{- end }}
```

Comment:
This comment looks like dead code. 

I don't fully understand how your code handles passing of parameters for the [index operator](https://zserio.org/doc/ZserioLanguageOverview.html#parameterized-types). Could you explain to me how your code handles this?

Reply:
This is a very clean solution and I have no concerns that it doesn't work! Thank you for this nice improvement!